e2e/app/admin/pauseportal.go

```diff
@@ -0,0 +1,71 @@
+package admin
+
+import (
+	"context"
+
+	"github.com/omni-network/omni/contracts/bindings"
+	"github.com/omni-network/omni/e2e/app"
+	"github.com/omni-network/omni/lib/errors"
+	"github.com/omni-network/omni/lib/log"
+
+	"github.com/ethereum/go-ethereum/common"
+)
+
+type PausePortalConfig struct {
+	Chain string // Pause a specific chain
+}
+
+var pausePortalABI = mustGetABI(bindings.PausePortalMetaData)
+
+func DefaultPausePortalConfig() PausePortalConfig {
+	return PausePortalConfig{
+		Chain: "",
+	}
+}
+
+func (cfg PausePortalConfig) Validate() error {
+	if cfg.Chain == "" {
+		return errors.New("chain is required")
+	}
+
+	return nil
+}
+
+// PausePortal pauses the portal contracts on a network. Only single chain is supported.
+func PausePortal(ctx context.Context, def app.Definition, cfg PausePortalConfig) error {
+	if err := cfg.Validate(); err != nil {
+		return errors.Wrap(err, "validate config")
+	}
+
+	s, err := setup(ctx, def)
+	if err != nil {
+		return errors.Wrap(err, "setup")
+	}
+
+	c, err := setupChain(ctx, s, cfg.Chain)
+	if err != nil {
+		return errors.Wrap(err, "setup chain")
+	}
+
+	if err := pausePortalForge(ctx, c.rpc, s.admin, c.PortalAddress); err != nil {
+		return errors.Wrap(err, "pause portal")
+	}
+
+	return nil
+}
+
+func pausePortalForge(ctx context.Context, rpc string, sender common.Address, portal common.Address) error {
+	calldata, err := pausePortalABI.Pack("run", portal)
```

Comment:
if we're using the ABI like this, rather than the script mode from forge, could we just have all of our admin actions in a single contract and share all the go boilerplate instead of having a separate go file per action?

Reply:
oh nvm, this is just encoding args

Reply:
No but we def can combine everything into a single script was thinking that too

Reply:
Just different entry points